fsdriver/driver.go

```diff
@@ -35,19 +52,58 @@ type Driver interface {
 	// NOTE(stevvooe): We may want to actually include the path manipulation
 	// functions here, as well. They have been listed below to make the
 	// discovery process easier.
-
-	// Join(path ...string) string
-	// IsAbs(string) bool
-	// Abs(string) (string, error)
-	// Rel(base, target string) (string, error)
+	Join(pathName ...string) string
+	IsAbs(pathName string) bool
+	Rel(base, target string) (string, error)
+	Base(pathName string) string
+	Dir(pathName string) string
+	Clean(pathName string) string
+	Split(pathName string) (dir, file string)
+	Separator() byte
+	NormalizePath(pathName string) string
+	// Abs(pathName string) (string, error)
 	// Walk(string, filepath.WalkFunc) error
 }
 
-func NewSystemDriver() (Driver, error) {
+// Unfortunately, os.File is a struct instead of an interface, an interface
+// has to be manually defined.
+var _ File = &os.File{}
+
+type File interface {
```

Comment:
I personally don't need those methods implemented. I could just have driver.Open fail for the LOW driver. However, it seems weird to me that there is a dependency on *os.File. Since *os.File isn't an interface, every implementation would have to eventually call os.Open to compile.